src/classes/CRLP_RecalculateBTN_CTRL.cls

```diff
@@ -130,4 +130,17 @@ public with sharing class CRLP_RecalculateBTN_CTRL {
         }
         return null;
     }
+
+    /*********************************************************************************************************
+    * @description Calculates the Soft Credit for the Contact.
+    * @return void.
+    */
+    private void calculateContactSoftCredit() {
```

Comment:
Do you think it's worth doing this as an asynchronous process? Right now, the logic will complete the Soft Credit rollup synchronously first and then kick off an async job to do the Hard Credits. 

I think these should be consistent - either both synchronous or both asynchronous. Open to discussing this though.

Reply:
Good point! I don't see the need to make the Hard Credit asynchronous since it is not a heavy process and it runs only on 1 Contact. If you don't see any downside, I can make the Hard Credit synchronous.

Reply:
I think you need to go the other direction - both asynchronous. Otherwise this will fail on any contact that has a decent number of attached hard credits + soft credits.